### PR TITLE
test(protocol fee): add edge case tests to protocol fee

### PR DESCRIPTION
**Pull request by jinoosss**
## Descriptions

This PR significantly enhances test coverage for the protocol fee module by adding comprehensive edge case tests across all core functionality. 

### New Test Files

- **api_test.gno**
  - Added tests for API endpoint functions
  - Covers version information and public API methods

- **assert_test.gno**
  - Added comprehensive authorization tests
  - Tests for `assertIsAdminOrGovStaker` with admin, gov/staker, and unauthorized user scenarios
  - Tests for `assertIsPoolOrPositionOrRouterOrStaker` with pool, position, router, staker, and unauthorized scenarios
  - Tests for `assertIsValidPercent` covering valid ranges (0%, 50%, 100%) and invalid inputs (negative, over limit)

- **getter_test.gno**
  - Added tests for all 10 getter functions
  - Covers `GetDevOpsPct`, `GetGovStakerPct` with various percentage configurations
  - Tests token list operations (`GetTokenListWithAmount`, `GetAmountOfToken`)
  - Tests accumulated transfer queries for both GovStaker and DevOps
  - Tests distribution history retrieval by token path

- **instance_test.gno**
  - Added instance creation tests
  - Tests for `NewProtocolFeeV1` with valid and nil store scenarios
  - Tests for `getProtocolFeeState` state access and default value verification

### Extended Test Files

- **protocol_fee_state_test.gno**
  - Added tests for state management public functions
  - Tests for `DevOpsPct`, `GovStakerPct` percentage calculations
  - Tests for AVL tree-based accumulated transfer tracking
  - Tests for token-specific transfer queries
  - Tests for `NewProtocolFeeStateBy` constructor

- **protocol_fee_test.gno**
  - Extended existing test suite with edge cases
  - Removed duplicate test function to avoid conflicts
  - Enhanced coverage for core protocol fee operations

## Test Coverage

This PR achieves comprehensive coverage for:

- Authorization and permission validation
- Percentage calculation and validation (0-10000 basis points)
- Token amount tracking and retrieval
- Accumulated transfer management for GovStaker and DevOps
- Distribution history tracking
- State initialization and access patterns

All tests follow table-driven testing patterns with clear test case descriptions covering success scenarios, edge cases, and error conditions.

### CODE REVIEW

File: contract/r/gnoswap/pool/v1/tick_test.gno [hunk 1]
```diff
@@ -758,3 +759,178 @@ func setMockPoolTick(
...
+			name:        "valid ticks in range",
+			tickLower:   -1000,
+			tickUpper:   1000,
+			shouldError: false,
+		},
+		{
+			name:        "valid ticks at MIN/MAX boundaries",
+			tickLower:   MIN_TICK,
+			tickUpper:   MAX_TICK,
+			shouldError: false,
```
Comment (junghoon-vans) [inline contract/r/gnoswap/pool/v1/tick_test.gno]: Duplicate with the following item: `tickLower at MIN, tickUpper at MAX`

**Review by junghoon-vans — APPROVED** [review]
